File: Ruby/tests/TurkeyHolidayValidator_test.rb
require 'minitest/autorun'
require_relative '../TurkeyHolidayValidator'

class TurkeyHolidayValidatorTest < Minitest::Test
  def setup
    @validator = TurkeyHolidayValidator.new
  end

  def test_turkey_fixed_holidays
    assert @validator.holiday?(Date.new(2024, 4, 23))
  end

  def test_turkey_islamic_holidays
    assert @validator.holiday?(Date.new(2024, 4, 10))
  end
end

File: Rust/tests/turkey_holiday_validator_test.rs
use crate::turkey_holiday_validator::TurkeyHolidayValidator;
use chrono::NaiveDate;

#[test]
fn test_turkey_fixed_holidays() {
    let validator = TurkeyHolidayValidator::new(false);
    assert!(validator.is_holiday(NaiveDate::from_ymd(2024, 4, 23)));
}

#[test]
fn test_turkey_islamic_holidays() {
    let validator = TurkeyHolidayValidator::new(false);
    assert!(validator.is_holiday(NaiveDate::from_ymd(2024, 4, 10)));
}
